src/conda_package_handling/streaming.py

```diff
@@ -13,30 +16,54 @@
 from . import exceptions
 
 
-def _extract(filename: str, dest_dir: str, components: list[str]):
-    """
-    Extract .conda or .tar.bz2 package to dest_dir.
-
-    If it's a conda package, components may be ["pkg", "info"]
-
-    If it's a .tar.bz2 package, components must equal ["pkg"]
-
-    Internal. Skip directly to conda-package-streaming if you don't need
-    exception compatibility.
-    """
-
+def _stream_components(
+    filename: str,
+    components: list[str],
+    dest_dir: str = "",
+) -> Generator[tuple[TarFile, TarInfo], None, None]:
```

Comment:
This is taken from [`conda-package-streaming`](https://github.com/conda/conda-package-streaming/blob/42510b1da4750467f2132bf1442b68ab8fd48680/conda_package_streaming/package_streaming.py#L71). Should we change it there too?

Reply:
I avoid `Generator` anytime the send type and return type are `None`, so I'm all in favor of making this change elsewhere too. Since this was done to follow the existing style it's up to you whether to include the change here or not.